src/annotator/config/index.js

```diff
@@ -1,14 +1,84 @@
 import settingsFrom from './settings';
 import { toBoolean } from '../../shared/type-coercions';
 
+/**
+ * List of allowed configuration keys per application context. Keys omitted
+ * in a given context will be removed from the relative configs when calling
+ * getConfig.
+ *
+ * @param {'sidebar'|'notebook'|'annotator'|'all'} [appContext] - The name of the app.
+ */
+function configurationContexts(appContext) {
+  switch (appContext) {
+    case 'annotator':
+      return ['clientUrl', 'showHighlights', 'subFrameIdentifier'];
+    case 'sidebar':
+      return [
+        'annotations',
+        'branding',
+        'enableExperimentalNewNoteButton',
+        'externalContainerSelector',
+        'focus',
+        'group',
+        'onLayoutChange',
+        'openSidebar',
+        'requestConfigFromFrame',
+        'services',
+        'showHighlights',
+        'sidebarAppUrl',
+        'theme',
+        'usernameUrl',
+      ];
+    case 'notebook':
+      return [
+        'branding',
+        'enableExperimentalNewNoteButton',
```

Comment:
The `enableExperimentalNewNoteButton` setting only makes sense in the sidebar because it is used in the "Page Notes" tab to create new notes. We don't currently support creating new annotations in the Notebook.